### PR TITLE
interfaces/dsp: add /dev/cavalry into dsp interface

### PR DESCRIPTION
**Pull request by woodrow-shen**
The /dev/cavalry is required to enable vector processor featuers when
flavor is from Ambarella chip.

Signed-off-by: Hsieh-Tseng Shen <woodrow.shen@canonical.com>

### CODE REVIEW

**Review by mvo5 — APPROVED** [review]
Code looks fine, I'm not sure we need a security review but I added Samuele so he can make that call. Please let me know if we should cherry pick this to 2.51.X

**Comment by woodrow-shen** [comment]
> Code looks fine, I'm not sure we need a security review but I added Samuele so he can make that call. Please let me know if we should cherry pick this to 2.51.X

@mvo5 Thank you. It would be great if you can cherry-pick this.

**Review by bboozzoo — APPROVED** [review]
LGTM

File: interfaces/builtin/dsp.go
Comment (anonymouse64) [inline interfaces/builtin/dsp.go:71]: before we merge this can you confirm the output of `udevadm info --query property /dev/cavalry` so we can confirm that this is the right rule to use? It probably is, I've never heard of anything else having a cavalry device node, but just to confirm.

Comment (woodrow-shen) [reply interfaces/builtin/dsp.go:71]: @anonymouse64 definitely agree, actually I did captured subsystem before I created this PR:

```
$ udevadm info --query property /dev/cavalry
DEVPATH=/devices/virtual/cavalry/cavalry
DEVNAME=/dev/cavalry
DEVMODE=0666
MAJOR=245
MINOR=0
SUBSYSTEM=cavalry
```

File: interfaces/builtin/dsp.go
Comment (pedronis) [inline interfaces/builtin/dsp.go:57]: I'm not sure what "the driver" refers to here?   it means the control device?  is just the node "for the CV2x VP" ?

File: interfaces/builtin/dsp.go
Comment (pedronis) [inline interfaces/builtin/dsp.go:58]: what "includes a set of device drivers" mean ?   are there multiple drivers all exposed through /dev/cavalry?

**Review by pedronis — COMMENTED** [review]
this looks alright but the comment block for /dev/cavalry is a bit opaque/confusing to  me

File: interfaces/builtin/dsp.go
Comment (anonymouse64) [inline interfaces/builtin/dsp.go:59]: To @pedronis' point, this comment is a bit confusing. I'm not sure if `/dev/cavalry` is itself a device node used to access low-level complexities and such on the vector processor or if it is actually a higher level thing and abstracts away the low-level complexities. In either case, it doesn't really matter for this file, I think we should just use something like this:

```suggestion
# The cavalry device node is used for managing the CV2x vector processor (VP).
```

File: interfaces/builtin/dsp.go
Comment (anonymouse64) [inline interfaces/builtin/dsp.go:54]: if you really want to include the CV2x / S6Lm DSP core information, I would rephrase this comment to keep closer to what was originally in the file with:

```suggestion
# The iav device node is the device node exposed for the specific IAV linux
# device driver used with the CV2x / S6Lm cores on an ambarella device
```

Comment (anonymouse64) [reply interfaces/builtin/dsp.go:54]: (obviously if there is only one core you can use `CV2x / S6Lm core`, I don't actually know if there is more than one or not

File: interfaces/builtin/dsp.go
Comment (anonymouse64) [reply interfaces/builtin/dsp.go:71]: thanks, the rule you provided for `KERNEL=="cavalry"` is fine then, thanks for confirming this

**Review by anonymouse64 — COMMENTED** [review]
some suggestions on wording of the comments in the apparmor file, but otherwise seems fine to me, thanks for adding this.

**Comment by woodrow-shen** [comment]
@pedronis @anonymouse64 Thanks for feedback. The comment blocks were based on Ambarella doc but it seems to cause some mixture of rough details between user space and drivers. Let's keep comments as simple as possible. Thanks.

**Review by anonymouse64 — APPROVED** [review]
lgtm, thanks for the comment simplifications

**Review by pedronis — APPROVED** [review]
thanks

**Pull request merged**
